File: src/main/java/me/alegian/thavma/impl/client/renderer/blockentity/PedestalBER.java
package me.alegian.thavma.impl.client.renderer.blockentity;

import me.alegian.thavma.impl.client.renderer.geo.layer.EmissiveGeoLayer;
import me.alegian.thavma.impl.common.block.entity.PedestalBE;
import net.neoforged.api.distmarker.Dist;
import net.neoforged.api.distmarker.OnlyIn;
import software.bernie.geckolib.model.DefaultedBlockGeoModel;
import software.bernie.geckolib.renderer.GeoBlockRenderer;

import static me.alegian.thavma.impl.ThavmaKt.rl;

@OnlyIn(Dist.CLIENT)
public class PedestalBER extends GeoBlockRenderer<PedestalBE> {
  public PedestalBER() {
    super(new DefaultedBlockGeoModel<>(rl("infusion_pedestal")));
    this.addRenderLayer(new EmissiveGeoLayer<>(this));
  }
}

File: src/main/java/me/alegian/thavma/impl/client/renderer/blockentity/PedestalBER.kt
package me.alegian.thavma.impl.client.renderer.blockentity

import me.alegian.thavma.impl.client.renderer.geo.layer.EmissiveGeoLayer
import me.alegian.thavma.impl.client.renderer.geo.layer.ItemRenderLayer
import me.alegian.thavma.impl.common.block.entity.PedestalBE
import me.alegian.thavma.impl.rl
import software.bernie.geckolib.model.DefaultedBlockGeoModel
import software.bernie.geckolib.renderer.GeoBlockRenderer


class PedestalBER : GeoBlockRenderer<PedestalBE>(DefaultedBlockGeoModel(rl("infusion_pedestal"))) {
  init {
    addRenderLayer(EmissiveGeoLayer(this))
    addRenderLayer(ItemRenderLayer(this, 1f))
  }
}
